From 3ca746545ccb848526ac152e614ce7ef0f37ac8f Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Fri, 5 Dec 2014 17:56:02 +0530
Subject: [PATCH 726/741] usb: gadget: Fix kernel boot dump for usb gadegt drv

commit 7a85ccf1638a1f0f7fb69a30ef17b61b6f2a790b from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Copy of_node property only for host mode so as to fix gadget mode
kernel dump messages

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/host/fsl-mph-dr-of.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index c1d25163d204..2f07115160f2 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -99,7 +99,8 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
-	pdev->dev.of_node = ofdev->dev.of_node;
+	if (pdata->operating_mode != FSL_USB2_DR_DEVICE)
+		pdev->dev.of_node = ofdev->dev.of_node;
 
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
-- 
2.17.1

